clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1798,6 +1818,11 @@ void handleFailure(Throwable throwable) {
 
     @Override
     public DescribeConfigsResult describeConfigs(Collection<ConfigResource> configResources, final DescribeConfigsOptions options) {
+        // avoid unnecessary metadata lookup
+        if (configResources.isEmpty()) {
```

Comment:
Below we break into `brokerResources` and  `unifiedRequestResources` and send two requests; we should check them and skip if possible respectively.